lua/rocks/fs.lua

```diff
@@ -36,6 +36,51 @@ function fs.file_exists(location)
     return false
 end
 
+--- Expand environment variables and tilde in the path string
+---@param path_str string
+---@return string
+function fs.expand_path(path_str)
```

Comment:
question: isn't this achievable with `vim.fn.expand`?

Reply:
Yeah, I was going to do that at first but the issue was that you would need to vim.schedule it to run in the main event loop since it's a vimscript function right? Then, you need to wait to get the result. Could use a future if it's in an async context but this would be needed for the config which is not async. Is there a better way to do it with the vim.fn.expand function?

Reply:
Hmm, I'll just make a `get_absolute_path_await` which wraps the original function with a future `set` and `wait` and call that one from the async context

Reply:
Decided to extract out a function for parsing the rocks_toml which is usable by both the config and helpers modules, hence, I kept the implementation of `expand_path` which doesn't depend on the main loop. Let me know if you'd still prefer that I use `vim.fn.expand`